Makefile

```diff
@@ -222,29 +223,43 @@ docker-hotrod:
 	docker build -t $(DOCKER_NAMESPACE)/example-hotrod:${DOCKER_TAG} ./examples/hotrod --build-arg TARGETARCH=$(GOARCH)
 
 .PHONY: run-all-in-one
-run-all-in-one:
+run-all-in-one: build-ui
 	go run -tags ui ./cmd/all-in-one --log-level debug
 
 .PHONY: build-ui
-build-ui:
+build-ui: cmd/query/app/ui/actual/gen_assets.go cmd/query/app/ui/placeholder/gen_assets.go
+	# Do nothing. If you need to force a rebuild of UI assets, run `make clean`.
+	#
+	# The `jaeger-ui` submodule contains the source code for the UI assets (requires Node.js 6+).
+	# The assets must be compiled first with `make build-ui`, which runs Node.js build and then
+	# packages the assets into a Go file that is `.gitignore`-ed.
+	#
+	# The packaged assets can be enabled by providing a build tag `ui`; for example:
+	# $ go run -tags ui ./cmd/all-in-one/main.go
```

Comment:
I think CONTRIBUTING file is a better place for this documentation than a makefile, nobody would look here, especially when reading instructions in CONTRIBUTING